src/doc/src/reference/config.md

```diff
@@ -657,6 +660,42 @@ Controls how often we display a notification to the terminal when a future incom
 * `always` (default): Always display a notification when a command (e.g. `cargo build`) produces a future incompat report
 * `never`: Never display a notification
 
+### `[gc]`
+
+The `[gc]` table defines settings for garbage collection in Cargo's caches, which will delete old, unused files.
+
+#### `[gc.auto]`
+
+The `[gc.auto]` table defines settings for automatic garbage collection in Cargo's caches.
+When running `cargo` commands, Cargo will automatically track which files you are using within the global cache.
+Periodically, Cargo will delete files that have not been used for some period of time.
+Currently it will delete files that have to be downloaded from the network if they have not been used in 3 months. Files that can be generated without network access will be deleted if they have not been used in 1 month.
+
+The automatic deletion of files only occurs when running commands that are already doing a significant amount of work, such as all of the build commands (`cargo build`, `cargo test`, `cargo check`, etc.), and `cargo fetch`.
+
+Automatic deletion is disabled if cargo is offline such as with `--offline` or `--frozen` to avoid deleting artifacts that may need to be used if you are offline for a long period of time.
+
+> **Note**: This tracking is currently only implemented for the global cache in Cargo's home directory.
+> This includes registry indexes and source files downloaded from registries and git dependencies.
+> Support for tracking build artifacts is not yet implemented, and tracked in [cargo#13136](https://github.com/rust-lang/cargo/issues/13136).
+>
+> Additionally, there is an unstable feature to support *manually* triggering garbage collection, and to further customize the configuration options.
+> See the [Unstable chapter](unstable.md#gc) for more information.
+
+##### `gc.auto.frequency`
+* Type: string
+* Default: `"1 day"`
+* Environment: `CARGO_GC_AUTO_FREQUENCY`
+
+This option defines how often Cargo will automatically delete unused files in the global cache.
+This does *not* define how old the files must be, those thresholds are described [above](#gcauto).
+
+It supports the following settings:
+
+* `"never"` --- Never deletes old files.
+* `"always"` --- Checks to delete old files every time Cargo runs.
+* An integer followed by "seconds", "minutes", "hours", "days", "weeks", or "months" --- Checks to delete old files at most the given time frame.
```

Comment:
“months” is an approximate number. 

https://github.com/rust-lang/cargo/blob/ea14e861c78cf5bb67e68ddeeea7ae9219a920a2/src/cargo/core/gc.rs#L373-L381

I can foresee someone will interpret `months` as monthly and think it will be clean on the same day when set, while it is not true especially in February. I don't think this is a thing we can't change after stabilization. Just calling it out if someone disagrees.

Reply:
Why not just remove months? It'd be counter-intuitive to anyone trying to use it...

Reply:
For myself, being able to say "6 months" is much easier than calculating out the number of days and reading the number of days.

Reply:
@epage But would *anyone* expect it to be 182.62125 days? Principle of least surprise...

Reply:
There is an inherent approximation when giving a unit.  The larger the unit, the larger the approximation.  If you say "6 months", you shouldn't care whether thats 180, 186, 182, or 182.62125.

btw laughing emoji's in a technical discussion like this come across as rude.

Reply:
> btw laughing emoji's in a technical discussion like this come across as rude.

Sigh... Here we go again. I was not even disagreeing with what you said. Just pointing out that it'd be surpirising for the user, as the OP of this thread has already pointed out (a different surprising aspect).

Reply:
> @epage But would _anyone_ expect it to be 182.62125 days? Principle of least surprise...

168 would be surprising; a bit over 180 not so much. The kind of surprise we're trying to avoid is purging data way earlier than expected.

Reply:
> For myself, being able to say "6 months" is much easier than calculating out the number of days and reading the number of days.

If the user goes to the trouble of customizing this in the config, I don't think having to calculate the number of days would be much of an extra hurdle. In effect, removing months would just simplify things with no real downside (and prevent future support questions where people are arguing over this again / trying to figure out what's going on with this approximation).

Reply:
A month equals 30 days is easier for me to accept